apps/docs/content/guides/realtime/authorization.mdx

```diff
@@ -0,0 +1,297 @@
+---
+id: 'authorization'
+title: 'Realtime Authorization'
+description: 'Authorization for Supabase Realtime'
+sidebar_label: 'Authorization'
+---
+
+You can control client access to Realtime [Broadcast](/docs/guides/realtime/broadcast) and [Presence](/docs/guides/realtime/presence) by adding Row Level Security policies to the `realtime.messages` table. Each RLS policy can map to a specific action a client can take:
+
+- Control which clients can broadcast to a Channel
+- Control which clients can receive broadcasts from a Channel
+- Control which clients can publish their presence to a Channel
+- Control which clients can receive messages about the presence of other clients
+
+<Admonition type="caution">
+
+Realtime Authorization is in Public Alpha. To use Authorization for your Realtime Channels, use `supabase-js` version `v2.44.0` or later.
+
+</Admonition>
+
+## How it works
+
+Realtime uses the `messages` table in your database's `realtime` schema to generate access policies for your clients when they connect to a Channel topic.
+
+By creating RLS polices on the `realtime.messages` table you can control the access users have to a Channel topic, and features within a Channel topic,
+
+The validation is done when the user connects. When their WebSocket connection is established and a Channel topic is joined, their permissions are calculated based on:
+
+- The RLS policies on tables in your `realtime` schema
+- The user information sent as part of their [Auth JWT](/docs/guides/auth/jwts)
+- The request headers
+- The Channel topic the user is trying to connect to
+
+When Realtime generates a policy for a client it does a test query on the `realtime.messages` table and then rolls it back. Realtime does not store any messages in your `realtime.messages` table.
+
+Using Realtime authorization involves two steps:
+
+- In your database, create an RLS policy on `realtime.messages`
+- In your client, instantiate the Realtime Channel with the `config` option `private: true`
+
+<Admonition type="caution">
+
+Increased RLS complexity can impact database performance and connection time, leading to higher connection latency and decreased join rates.
+
+</Admonition>
+
+## Helper functions
+
+You can use the following helper functions when writing RLS policies:
+
+### `realtime.topic`
+
+Returns the Channel topic the user is attempting to connect to.
+
+```sql
+create policy "authenticated users can read all messages in room_1"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  (select realtime.topic()) = 'room_1'
+);
+```
+
+## Examples
+
+The following examples use this schema:
+
+```sql
+create table public.profiles (
+  email text not null,
+  user_id uuid references auth.users (id)
+);
+
+create table public.rooms_users (
+  topic text references realtime.messages (topic),
```

Comment:
@chasers `realtime.messages` isn't storing anything so can't do this reference.

can you update the example SQL to match https://github.com/supabase/supabase/tree/master/examples/realtime/nextjs-authorization-demo#create-tables?

Reply:
Sure